File: java/2D_Arrays/StaircaseSearch.java
public class StaircaseSearch {

    public static boolean staircaseSearch(int[][] matrix, int key) {
        int row = matrix.length-1, col = 0;

        while (row >= 0 && col < matrix[0].length) {
            if (matrix[row][col] == key) {
                System.out.println("found key at (" + row + "," + col + ")");
                return true;
            } else if (key < matrix[row][col]) {
                row--;
            } else {
                col++;
            }
        }
        System.out.println("key not found");
        return false;
    }

    public static void main(String[] args) {

        int[][] matrix = { { 10, 20, 30, 40 },
                { 15, 25, 35, 45 },
                { 27, 29, 37, 48 },
                { 32, 33, 39, 50 } };

        int key = 33;
        staircaseSearch(matrix, key);

    }
}

File: python/2D_Arrays/StaircaseSearch.py
def staircaseSearch(matrix, key):
    row = len(matrix)-1
    col = 0

    while (row >= 0 and col < len(matrix[0])):
        if (matrix[row][col] == key):
            print("found at (" + str(row) + "," + str(col) + ")")
            return True
        elif(key < matrix[row][col]):
            row-=1
        else:
            col+=1
    
    print("key not found")
    return False

matrix = [[10, 20, 30, 40], [15, 25, 35, 45], [27, 29, 37, 48], [32, 33, 39, 50]]
staircaseSearch(matrix, 33)